packages/components/src/tree-grid/types.ts

```diff
@@ -0,0 +1,116 @@
+export type TreeGridRowProps = {
+	/**
+	 * The children to be rendered in the row.
+	 */
+	children: React.ReactNode;
+	/**
+	 * An integer value designating the level in the hierarchical tree structure.
+	 * Counting starts at 1. A value of `1` indicates the root level of the structure.
+	 */
+	level: React.AriaAttributes[ 'aria-level' ];
+	/**
+	 * An integer value that represents the position in the set.
+	 * A set is the count of elements at a specific level. Counting starts at 1.
+	 */
+	positionInSet: React.AriaAttributes[ 'aria-posinset' ];
+	/**
+	 * An integer value that represents the total number of items in the set,
+	 * at this specific level of the hierarchy.
+	 */
+	setSize: React.AriaAttributes[ 'aria-setsize' ];
```

Comment:
It looks like types from `React.AriaAttributes` can also be `undefined` — should we wrap them in `NonNullable` here, since these props are not optional ?

Reply:
Good call 👍 37031c57d72b59b8b6a285cab92c5e2c23d78c15